### DISCUSSION TITLE
XCode error gradle -p ../Android launchDebug

### DISCUSSION THREAD

**Discussion by markonikolov16** [skiptools/skip#157] [General]
My Xcode build always fails with: 
The gradle command failed. Review the log for details and consult https://skip.tools/docs/faq for common solutions. Command: gradle -p ../Android launchDebug
Yet when I run the command directly in the console it works fine and the Android app starts working on the simulator.

Checkup passes successfully.

**Comment by marcprux** [comment]
Can you post the log file and/or screenshot of the failure when running from Xcode?

**Comment by markonikolov16** [comment]
![Screenshot 2024-06-26 at 5 15 20 PM](https://github.com/skiptools/skip/assets/84811937/0806b817-7737-46d4-bfe3-f518d8e0347e)

There is no log file as checkup passes fine. 
Init also passes fine.

**Comment by marcprux** [comment]
There are two surprising errors in there.

The first, from the Xcode build errors

> Undefined symbol: _main

I'm not sure where this would be coming from. Are you able to build and run other iOS apps (e.g., one that you just create using the Xcode project creator)?

And then this:

> ERROR: JAVA_HOME is set to an invalid directory: /opt/homebrew/opt/java

Can you let me know if that folder exists, by checking it on the Terminal? And can you let me know if you are on an ARM or Intel Mac?

**Reply by markonikolov16** [reply]
The symbol: _main and linker errors go away after a clean so you can ignore them.
The directory does not exist. Java_home is set to /Library/Java/JavaVirtualMachines/jdk-22.jdk/Contents/Home and its properly processed during checkup.
Im using M1 so ARM.

**Comment by marcprux** [comment]
Did you install Skip with homebrew (i.e., `brew install skiptools/skip/skip`)? That should have added a link to /opt/homebrew/opt/java for the OpenJDK installation that Skip will use.

Does the /opt/homebrew/ folder exist at all? If not, is it possible that you upgraded your system from a previous Intel machine (in which case Homebrew would be located in `/usr/local/ – the remedy for this is to re-install homebrew from [brew.sh](https://brew.sh)).

Also, can you post the output of the terminal command `env`?

As a temporary workaround, you may be able to set your JAVA_HOME in your `~/.zshrc` file and then re-start Xcode, but this should be unnecessary and we would like to figure out the cause of the problem.

**Reply by markonikolov16** [reply]
Reinstalling homebrew to the proper location and resetting Java home fixed the issue. Thank you
